modules/quantcastBidAdapter.js

```diff
@@ -24,31 +25,20 @@ export const QUANTCAST_FPA = '__qca';
 export const storage = getStorageManager(QUANTCAST_VENDOR_ID, BIDDER_CODE);
 
 function makeVideoImp(bid) {
-  const video = {};
-  if (bid.params.video) {
-    video['mimes'] = bid.params.video.mimes;
-    video['minduration'] = bid.params.video.minduration;
-    video['maxduration'] = bid.params.video.maxduration;
-    video['protocols'] = bid.params.video.protocols;
-    video['startdelay'] = bid.params.video.startdelay;
-    video['linearity'] = bid.params.video.linearity;
-    video['battr'] = bid.params.video.battr;
-    video['maxbitrate'] = bid.params.video.maxbitrate;
-    video['playbackmethod'] = bid.params.video.playbackmethod;
-    video['delivery'] = bid.params.video.delivery;
-    video['placement'] = bid.params.video.placement;
-    video['api'] = bid.params.video.api;
-  }
-  if (bid.mediaTypes.video.mimes) {
-    video['mimes'] = bid.mediaTypes.video.mimes;
+  const videoInMediaType = utils.deepAccess(bid, 'mediaTypes.video') || {};
+  const videoInParams = utils.deepAccess(bid, 'params.video') || {};
+  const video = Object.assign({}, videoInParams, videoInMediaType);
+
+  if (video.playerSize) {
+    video.w = video.playerSize[0];
+    video.h = video.playerSize[1];
   }
-  if (utils.isArray(bid.mediaTypes.video.playerSize[0])) {
-    video['w'] = bid.mediaTypes.video.playerSize[0][0];
-    video['h'] = bid.mediaTypes.video.playerSize[0][1];
-  } else {
-    video['w'] = bid.mediaTypes.video.playerSize[0];
-    video['h'] = bid.mediaTypes.video.playerSize[1];
+  for (const prop of VIDEO_PROPS_TO_REMOVE) {
```

Comment:
a bit of a nit-pick - would it make more sense to have an allow list of properties rather than a block list? otherwise the video object on mediaTypes or params could introduce new properties and we wouldn't filter them.

Reply:
done